### PR TITLE
Mark sync_state_genSyncSpec JSON-RPC as safe

### PR DESCRIPTION
**Pull request by tomaka**
The `sync_state_genSyncSpec` JSON-RPC function returns the chain specification currently running on the node, with a checkpoint.

I don't really understand why it is marked as "unsafe".
The JSON-RPC function was added in https://github.com/paritytech/substrate/pull/7225, and was already marked as "unsafe" from the start, and there was no discussion about this.

Having it safe would make our lives a bit easier, as it would now be possible to download checkpoints from public JSON-RPC nodes, rather than having to run a separate node just for that.

polkadot companion: https://github.com/paritytech/polkadot/pull/4918

### CODE REVIEW

**Comment by bkchr** [comment]
The polkadot chain spec is for example 9MB. It would mean we would need to send every time 9MB of data someone calls this endpoint. I know, that you can also achieve the same using other endpoints. 

Why do you not just use the internal RPC nodes?

**Comment by tomaka** [comment]
> Why do you not just use the internal RPC nodes?

Well, the pragmatic reason is I'd like to write a bot that automatically grabs chain specs, and having to be within the VPN (or in general needing credentials) makes this way more complicated than it could be.
If we don't merge this PR, I would actually instead implement generating checkpoints from smoldot, because that's going to be easier than obtaining and storing credentials to our internal stuff.

It also makes it easier when a parachain team sends us a link to one of their nodes, as you could just grab the chain spec from this node.
In other words, a RPC node could become a trusted place where to fetch the spec of a chain.

File: client/sync-state-rpc/src/lib.rs [hunk 1]
```diff
@@ -143,17 +142,20 @@ where
...
+	///
+	/// The `deny_unsafe` parameter is deprecated and unused. It is there only to maintain
+	/// API backwards compatibility.
 	pub fn new(
 		chain_spec: Box<dyn sc_chain_spec::ChainSpec>,
 		client: Arc<Backend>,
 		shared_authority_set: SharedAuthoritySet<Block>,
 		shared_epoch_changes: SharedEpochChanges<Block>,
-		deny_unsafe: sc_rpc_api::DenyUnsafe,
+		_deny_unsafe: sc_rpc_api::DenyUnsafe,
```
Comment (bkchr) [inline client/sync-state-rpc/src/lib.rs:153]: I would remove this.

**Review by bkchr — APPROVED** [review]

**Comment by tomaka** [comment]
Just want to point out that if we need Cumulus companion, this will have to wait for next week because this week I only have mobile Internet and around 5 GiB left for the month.

**Review by bkchr — APPROVED** [review]

**Comment by bkchr** [comment]
bot merge
